source/ranges.tex

```diff
@@ -322,15 +322,10 @@
 
 \item
   Otherwise, \tcode{\placeholdernc{decay-copy}(begin(E))} if it is a
-  valid expression and its type \tcode{I} models
-  \libconcept{input_or_output_iterator} with overload
-  resolution performed in a context that includes the declarations:
-  \begin{codeblock}
-  template<class T> void begin(T&&) = delete;
-  template<class T> void begin(initializer_list<T>&&) = delete;
-  \end{codeblock}
-
-  and does not include a declaration of \tcode{ranges::begin}.
+  valid expression
+  when `begin` is looked up using only argument-dependent lookup\iref{basic.lookup.argdep}
+  and its type \tcode{I} models
+  \libconcept{input_or_output_iterator}.
```

Comment:
Many of these ranges customization points end with the formula "...and its type `I` models `some-concept`." I don't understand why we need to introduce the name `I`, given that we never use the name `I` again in this scope. However, I didn't change these formulas in this patch.

Reply:
I believe that's because [[customization.point.object]p5](http://eel.is/c++draft/library#customization.point.object-5) requires it, although some CPOs don't do that, IIRC.

Reply:
I didn't understand this answer, btw. My guess is that you were answering the unasked question "Why does the formula mention that the type models `some-concept`?", but not "Why does the formula introduce the name `I` to refer to a type that is never mentioned again?"

I mean, it seems like we should just say "...and its type models `some-concept`." Interjecting the unused identifier `I` seems unnecessary.